src/component/ItemDetails.js

```diff
@@ -20,20 +40,41 @@ const ItemDetails = ({ token, purchased }) => {
         }
 
         // Below lines calculate next purchase date using dayjs 'add' function.
-        const lastPurchaseDate = item.dateOfPurchase.toDate();
-        const nextPurchaseDate = dayjs(lastPurchaseDate)
-          .add(item.numberOfDays.toString(), 'day')
-          .toDate();
+        let lastPurchaseDate;
+        let nextPurchaseDate;
+
+        if (item.dateOfPurchase) {
+          lastPurchaseDate = item.dateOfPurchase.toDate();
+          nextPurchaseDate = dayjs(lastPurchaseDate)
+            .add(item.numberOfDays.toString(), 'day')
+            .toDate();
+
+          setIsNewItem(false);
+        }
 
         return (
           <main>
             <BackButton />
             <h1>{item.name}</h1>
             <ul className="itemDetails">
-              <li>Last purchase: {lastPurchaseDate.toDateString()}</li>
-              <li>Next purchase: {nextPurchaseDate.toDateString()}</li>
+              <li>
+                Last purchase:{' '}
+                {isNewItem ? 'None' : lastPurchaseDate.toDateString()}
+              </li>
+              <li>
+                Next purchase:{' '}
+                {isNewItem ? 'None' : nextPurchaseDate.toDateString()}
+              </li>
```

Comment:
Thanks for taking this a step further so users can actually view item details when item has never been purchased.

Reply:
Yeah! I realized if we had a brand new item, we wouldn't be able to delete it, because we couldn't get to the item details page where the delete button is. @ClaudiaStar